animation_nodes/utils/midi.py

```diff
@@ -0,0 +1,101 @@
+from ..libs.midiparser.parser import MidiFile
```

Comment:
Can you explain how tempo maps works?

Reply:
We need to get the MIDI events at the right time in seconds
tempoMap contains a table of the tempos encountered in a track as well as the translation in seconds of the moments when the tempos change.
The timeMap.getRealtime method converts a position of type timeInTicksCumul into real seconds knowing the changes in tempos that affect the ticks versus seconds relationship

I hope I have been clear enough.
tempoMap makes it possible to simplify the ticks versus seconds conversion by having prepared the necessary information when loading the MIDI file rather than on the fly then by recalculating.

Reply:
Yes. That's clear. I will probably request changes on this part of the code after the first review pass is done.

Reply:
Yes I think there are surely other ways to do it